cmd/boot-aws/main.go

```diff
@@ -192,6 +192,13 @@ func doSetup(a *awscloud.AWS, filename string, flags *pflag.FlagSet, res *resour
 		return err
 	}
 
+	var importRolePtr *string
+	if importRole, err := flags.GetString("import-role"); importRole != "" {
```

Comment:
(nitpick) Maybe the slightly more canoical:
```go
	var importRolePtr *string
	importRole, err := flags.GetString("import-role")
	if err != nil {
		return err
	}
	if importRole != "" {
		importRolePtr = &importRole
	}
```
ie. error check first? I know it's the same but when reading go code anything that diverges from this pattern automatically stops my flow.

Reply:
I agree with errors first but I do like the limited scoping you get from the `if ret, err := func(); err != nil ...`

Reply:
What about extracting a helper function to get both error handling first, and limiting the scope? See my latest force-push.

Reply:
Well, Michael approved, so I'm resolving this and merging.